tests/state_test_util.go

```diff
@@ -335,16 +336,17 @@ func MakePreState(db database.DBManager, accounts blockchain.GenesisAlloc, isTes
 	sdb := state.NewDatabase(db)
 	statedb, _ := state.New(common.Hash{}, sdb, nil, nil)
 	for addr, a := range accounts {
-		if len(a.Code) != 0 {
-			if _, ok := types.ParseDelegation(a.Code); ok {
-				statedb.SetCodeToEOA(addr, a.Code, rules)
-			} else {
-				if isTestExecutionSpecState {
-					statedb.CreateSmartContractAccount(addr, params.CodeFormatEVM, rules)
-				}
-				statedb.SetCode(addr, a.Code)
+		if isTestExecutionSpecState {
+			if _, ok := types.ParseDelegation(a.Code); (ok && rules.IsPrague) || (len(a.Code) == 0 && len(a.Storage) != 0) {
+				statedb.CreateEOA(addr, false, accountkey.NewAccountKeyLegacy())
+			} else if len(a.Code) != 0 {
+				statedb.CreateSmartContractAccount(addr, params.CodeFormatEVM, rules)
 			}
 		}
+
+		if len(a.Code) != 0 {
+			statedb.SetCode(addr, a.Code)
```

Comment:
I'm not sure, but does EOA with delegation code work? Don't we need to call `SetCodeToEOA` somewhere?